### ISSUE TITLE
HTTPMethods links in doc are broken

### ISSUE THREAD

**Issue by jgravois** [Esri/arcgis-rest-js#86]
when i reviewed https://github.com/Esri/arcgis-rest-js/commit/32face15dc6da19302e9decd32f8650e7ea0869b i noticed that i overwrote some links to nowhere with actual '404's.

everything _looks_ okay in the `typedoc.json`, but we aren't yet linking correctly to [`HTTPMethods`](https://esri.github.io/arcgis-rest-js/api/request/HTTPMethods/).

**Comment by jgravois** [comment]
> everything looks okay in the typedoc.json

that's not technically correct. i took a deeper look at this and we aren't generating a `src` or `pageUrl` for **inherited** properties in the typedoc.json.

example:
[`IAuthenticatedRequestOptions`](https://esri.github.io/arcgis-rest-js/api/auth/IAuthenticatedRequestOptions/)

```json
{
    "id": 496,
    "name": "httpMethod",
    "kind": 1024,
    "kindString": "Property",
    "flags": {
    "isExported": true,
    "isOptional": true
    },
    "comment": {
    "shortText": "The HTTP method to send the request with."
    },
    "sources": [
    {
        "fileName": "arcgis-rest-auth/node_modules/@esri/arcgis-rest-request/dist/esm/request.d.ts",
        "line": 28,
        "character": 14
    }
    ],
    "type": {
    "type": "reference",
    "name": "HTTPMethods",
    "id": 88
    },
    "inheritedFrom": {
    "type": "reference",
    "name": "IRequestOptions.httpMethod"
    },
    "icon": "tsd-kind-property tsd-parent-kind-interface"
}
```

**Comment by jgravois** [comment]
resolved via #93
